include/xmipp4/core/backend.hpp

```diff
@@ -57,9 +58,9 @@ class XMIPP4_CORE_API backend
     /**
      * @brief Get the name of the backend.
      * 
-     * @return const std::string& The name.
+     * @return std::string The name.
      */
-    virtual const std::string& get_name() const noexcept = 0;
```

Comment:
Question, why not const? it's not meant to be modified

Reply:
Thats something upto the caller

Reply:
The method is returning a copy of the name. The caller should decide to make it const or not

Reply:
Formerly, it was declared as const std::string& because the method was returning a reference. Thus, all callers would receive the same actual string. If it were not declared as const and one of them were to modify it, it would modify it for all all.

However now that we are returning by value (a copy), it is usual practice to return without any `const`/`volatile` qualifications.

Reply:
https://www.reddit.com/r/cpp_questions/comments/13xuu37/const_correctness_and_const_return_values/?rdt=63613